server/src/main/java/io/druid/client/selector/ServerSelectorStrategy.java

```diff
@@ -32,5 +34,7 @@
 })
 public interface ServerSelectorStrategy
 {
-  public QueryableDruidServer pick(Set<QueryableDruidServer> servers, DataSegment segment);
+  QueryableDruidServer pick(Set<QueryableDruidServer> servers, DataSegment segment);
+
+  List<QueryableDruidServer> pick(Set<QueryableDruidServer> servers, DataSegment segment, int numServersToPick);
```

Comment:
oh, sorry, misunderstood for which line this comment is.
yes, additional method will break the compatibility but I don't think that there are many custom implementations for ServerSelectorStrategy and TieredSelectorStrategy, though I don't know for sure, so it's better to ask this question: do you think it's critical and we can not extend this interfaces? 
if yes, of course, we can leave them as is and use current approach to pick multiple candidates but from my perspective it does not look good both from design and performance perspective.